app/src/ui/repository.tsx

```diff
@@ -241,12 +243,51 @@ export class RepositoryView extends React.Component<
     }
   }
 
+  private currentStashForBranch() {
+    const { branchesState, stashEntries } = this.props.state
+    const tip = branchesState.tip
+    if (tip.kind !== TipState.Valid) {
+      return null
+    }
+
+    // Using the short form ref name because the branch model does not keep track
+    // of the canonical ref name. Todo: update model to use the canonical ref name.
+    const branch = tip.branch
+    const stashEntry = stashEntries.get(branch.name)
+    if (stashEntry === undefined) {
+      return null
+    }
+
+    return stashEntry
+  }
+
   private renderContent(): JSX.Element | null {
     const selectedSection = this.props.state.selectedSection
-
     if (selectedSection === RepositorySectionTab.Changes) {
       const { changesState } = this.props.state
-      const { workingDirectory, selectedFileIDs, diff } = changesState
+      const {
+        workingDirectory,
+        selectedFileIDs,
+        diff,
+        isShowingStashEntry,
+      } = changesState
+
+      if (isShowingStashEntry && selectedFileIDs.length === 0) {
```

Comment:
There is no need to place this block behind the feature flag because `currentStashForBranch` uses  the feature flag in its implementation.